### PR TITLE
i18n: Add isRTL function

### LINKED ISSUE #19212: i18n: the isRTL flag should be provided by the current locale data
Currently, when Gutenberg code needs to know if the current text direction is LTR or RTL, it gets it from various sources:
- the Block Editor has `isRTL` value in its settings
- components query the DOM: `document.documentElement.dir === 'rtl'`

But there is one place that could and should be the source of truth for everyone -- the `@wordpress/i18n` library and its current locale.

Every WordPress locale has a special translation string called `_x( 'ltr', 'text direction' )`. If the translation is `rtl`, then we're dealing with a RTL language. See the usage of this translation string in [Core PHP `WP_Locale` class](https://github.com/WordPress/WordPress/blob/master/wp-includes/class-wp-locale.php#L217)

`@wordpress/i18n` should export a new function that returns authoritative value of the RTL flag:
```js
import { isRTL } from `@wordpress/i18n`;

const popoverAlignment = isRTL() : 'left' : 'right';
```

### PR DESCRIPTION
**Pull request by sirreal**
## Description
Add `isRTL` function for checking whether the locale is right-to-left.

This follows the implementation of `WP_Locale::is_rtl()`.

https://developer.wordpress.org/reference/classes/wp_locale/is_rtl/

Part of #19212.

The unit tests have been slightly restructured to provide isolation of the `i18n` module between tests.

## How has this been tested?

Includes unit tests.

## Types of changes
New feature (non-breaking change which adds functionality)

Add `isRTL` function to `@wordpress/i18n`.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/i18n/README.md [hunk 1]
```diff
@@ -27,6 +27,14 @@ For a complete example, see the [Internationalization section of the Block Edito
 
 <!-- START TOKEN(Autogenerated API docs) -->
 
+<a name="isRTL" href="#isRTL">#</a> **isRTL**
+
+Check if current locale is RTL.
```
Comment (simison) [inline packages/i18n/README.md]: Explaining what "RTL" means could be good here.

Comment (sirreal) [reply packages/i18n/README.md]: This is the exact documentation string from `WP_Locale::is_rtl()`, but you're right 🙂 

I'll add some detail.

**Review by simison — APPROVED** [review]
Code looks good, thanks for adding docs and improving tests!

I haven't tested this in any way other than confirming that tests run.

**Comment by youknowriad** [comment]
Interesting way of checking the rtl direction :) 
We have a few different ways of doing it already in the code base, do you think we should update our existing checks to use this new function?

**Comment by sirreal** [comment]
> Interesting way of checking the rtl direction :)

It seems to be the canonical way of specifying direction in WordPress. See https://github.com/WordPress/WordPress/blob/511a297fb4b89ad32e89ea0373347cf92367e1c3/wp-includes/class-wp-locale.php#L217-L219

> We have a few different ways of doing it already in the code base, do you think we should update our existing checks to use this new function?

That's what https://github.com/WordPress/gutenberg/issues/19212 suggests. I agree that there's value in having a single, well-defined way of checking the text direction.

File: packages/i18n/README.md (hunk 1 above)
Comment (sirreal) [reply packages/i18n/README.md]: Added RTL explanation courtesy of https://developer.mozilla.org/en-US/docs/Glossary/rtl.

**Comment by simison** [comment]
> We have a few different ways of doing it already in the code base, do you think we should update our existing checks to use this new function?

Spotted here:

https://github.com/WordPress/gutenberg/blob/61e936cee1209fe9f5d020827af031d2a166bd64/packages/components/src/utils/rtl.js#L6-L8

**Comment by youknowriad** [comment]
yes, we also have a setting in the block-editor store I believe and another check in the Popover component.

**Comment by simison** [comment]
@youknowriad would you expect to see those uses updated in this PR or good for follow-ups?

**Comment by youknowriad** [comment]
I'm totally fine if we merge this as is :) 

But I've been promised follow-ups in the past and sometimes they just don't happen :P

**Comment by sirreal** [comment]
I'll merge this in the next few days unless there are issues or there's a good reason to wait.

I didn't plan to modify anything beyond `@wordpress/i18n` with this, but I can certainly help to migrate existing functionality to use `isRTL`.

**Comment by aduth** [comment]
Arriving to this late since I'm only recently returning from AFK, but seems like a reasonable addition to me 👍

**Comment by sirreal** [comment]
> We have a few different ways of doing it already in the code base, do you think we should update our existing checks to use this new function?

I started to look at this and have some concerns. There's currently a mix of reactive (re-renders as necessary) and non-reactive RTL lookups.

For example, this "hook" looks reactive, but isn't:

https://github.com/WordPress/gutenberg/blob/a46cd2fe66c3a5aab460b09275a970d919c7baf0/packages/components/src/utils/rtl.js#L6-L15

This _is_ reactive, but doesn't know anything about RTL and is reading a value from a data store:

https://github.com/WordPress/gutenberg/blob/67e2df7275de5972f9aca83bceabbed6ae396b81/packages/block-library/src/paragraph/edit.js#L37-L39

Then there are straightforward cases like this:

https://github.com/WordPress/gutenberg/blob/a46cd2fe66c3a5aab460b09275a970d919c7baf0/packages/components/src/range-control/rail.js#L51

---

Replacing the reactive RTL setting lookups with `isRTL` at this time seems undesirable. Would the `isRTL` setting disappear from `block-editor`?

https://github.com/WordPress/gutenberg/blob/320f460db91e1f0acbf91d45d6a85509e3ca5d85/lib/edit-site-page.php#L65-L71

I'd like to see RTL information come from the locale and be updated in a reactive way. Perhaps is worth waiting for #19210 to be resolved?

**Comment by youknowriad** [comment]
> Replacing the reactive RTL setting lookups with isRTL at this time seems undesirable. Would the isRTL setting disappear from block-editor?

I think it's fine, we just need to add a dev note about it. It's unlikely that anyone aside us is using that setting right now.

**Comment by sirreal** [comment]
I've started some work on using `isRTL` throughout Gutenberg: https://github.com/WordPress/gutenberg/pull/22806

**Pull request merged**